### ISSUE TITLE
Code custom error analysis functions and capabilities

### ISSUE THREAD

**Issue by mayhemsloth** [mayhemsloth/Drum-Tabber#22]
Now, with the ability to train a model using a custom model training loop, I need to have some metrics to determine which models are the "best" when comparing across different types of models, besides the validation loss error. Specifically because the dataset is heavily imbalanced (most drum events are sparse compared to the duration of an entire song, so most spectro windows will have labels of 0 for everything), I need to code functions for better metrics to ensure that the model isn't just biased towards guessing 0 for everything (or at least the truly sparse events like tom hits). 

Additionally, for metrics like true positives and true negatives, and false positives and false negatives, I need to decide what that _actually_ means. For example, when creating the targets, I currently label a range of spectro windows as the ground truth, based on one sample number. What should count as a true positive then? That the model correctly produced a *peak* around that sample number/general location? Because ultimately that *is* what I am looking for: a model that can output the drum **tab**, not necessarily the exact 10 ms window that a drum onset event occurs. 

This issue is resolved when I have a various number of metrics automatically calculated in the training loop, so that I know upon each song or validation song perhaps what the model is doing and how it is behaving at a deeper level than just the loss. This may depend on the existence of predictions-to-tab functions, depending on how the metrics are defined.

**Comment by mayhemsloth** [comment]
This is almost done, but I need to rectify the batch size increase when it hits the end of the song. To make batch sizes equal in the last section, I took the beginning of the input array and appended it so that the last batch is equal. Effectively this creates more peaks in the prediction than actually exist in the label_ref_df. You need to equivalently append some amount of rows to the label_ref_df to account for this.
